CONTRIBUTING.md

```diff
@@ -107,7 +121,47 @@ just ready
 This command will run the same commands of the CI: format, lint, tests and code generation.
 Eventually everything should be "green" 🟢 and commit all the code that was generated.
 
-## Language Server and VS Code Extension Development
+#### Generated files
+
+If you work on some parser and you create new nodes or modify existing ones, will need to run a command to update some files that are auto-generated.
+
+##### `cargo codegen grammar`
+
+This command will update the syntax of the parsers.
+
+The source is generated from the [`ungram` files](https://github.com/biomejs/biome/blob/main/xtask/codegen/js.ungram).
+
+##### `cargo codegen test`
+
+This command will create new tests for your parser. We currently have a neat infrastructure
+where tests for parser are generated com inline comments found inside
+the source code.
+
+
+##### `cargo codegen analyzer`
+
+This command will detect linter rules declared in the `analyzers` and `assists` directories in `biome_analyze`, regenerate the index modules `analyzers.rs` and `assists.rs` to import these files, and update the registry builder function in `registry.rs` to include all these rules.
+It will also regenerate the configuration of the rules.
+
+##### `cargo coverage`
+
+This command will check and report parser conformance against different test suites.
+We currently target the [Official ECMAScript Conformance Test Suite](https://github.com/tc39/test262) and
+the [Typescript Test Suite](https://github.com/microsoft/TypeScript/tree/main/tests)
+
+The test suites are included as git submodules and can be pulled using:
+
+```bash
+git submodule update --init --recursive
+```
+
+### crate dependencies
+
+[Workspace dependencies](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table) are used, and many dependencies are defined in Cargo.toml in the root.
+
+Internal crates are loaded with `workspace = true` for each crate. About dev-dependencies, we use [path dependencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-path-dependencies) to avoid requiring these crates to be published when doing cargo publish.
```

Comment:
```suggestion
Internal crates are loaded with `workspace = true` for each crate. About `dev-dependencies`, we use [path dependencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-path-dependencies) to avoid requiring the published version of these crates.
```